### PR TITLE
Refactoring Recycler with databinding

### PR DESCRIPTION
**Pull request by tolia-ogurt**

### CODE REVIEW

File: app/src/main/java/com/example/trempel/BindingAdapters.kt [hunk 1]
```diff
@@ -52,28 +50,14 @@ fun View.setVisibility(isVisible: Boolean) {
...
-}
-
-private fun getOrCreateCategoriesAdapter(recyclerView: RecyclerView): CategoriesAdapter {
-    val categoriesAdapter = CategoriesAdapter()
-    recyclerView.adapter = categoriesAdapter
-    return categoriesAdapter
+@BindingAdapter("items")
+fun setRecyclerViewItems(recyclerView: RecyclerView, items: List<RecyclerItem>?) {
+    var adapter = (recyclerView.adapter as? DataBindingRecyclerAdapter)
+    if (adapter == null) {
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/BindingAdapters.kt:56]: can we use any scopeFunction?

File: app/src/main/java/com/example/trempel/ui/DataBindingRecyclerAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package com.example.trempel.ui
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+
+class DataBindingRecyclerAdapter :
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/ui/DataBindingRecyclerAdapter.kt]: `internal`?

File: app/src/main/java/com/example/trempel/ui/DiffCallBack.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package com.example.trempel.ui
+
+import androidx.recyclerview.widget.DiffUtil
+
+class DiffCallback : DiffUtil.ItemCallback<RecyclerItem>() {
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/ui/DiffCallBack.kt]: `internal`?

File: app/src/main/java/com/example/trempel/ui/RecyclerItem.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.example.trempel.ui
+
+import androidx.annotation.LayoutRes
+import androidx.databinding.ViewDataBinding
+
+data class RecyclerItem(
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/ui/RecyclerItem.kt]: `internal`?

File: app/src/main/java/com/example/trempel/ui/RecyclerItemComparator.kt [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package com.example.trempel.ui
+
+interface RecyclerItemComparator {
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/ui/RecyclerItemComparator.kt]: `internal`?

File: app/src/main/java/com/example/trempel/ui/categories/CategoriesItemViewModel.kt [hunk 6]
```diff
@@ -4,13 +4,27 @@ import android.view.View
 import androidx.lifecycle.ViewModel
 import androidx.navigation.findNavController
 import com.example.trempel.network.CategoryDomainModel
+import com.example.trempel.ui.RecyclerItemComparator
 
 internal class CategoriesItemViewModel(
     val category: CategoryDomainModel
-) : ViewModel() {
+) : ViewModel(),RecyclerItemComparator {
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/ui/categories/CategoriesItemViewModel.kt]: space is required after `,`

Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/ui/categories/CategoriesItemViewModel.kt]: done

File: app/src/main/java/com/example/trempel/ui/RecyclerItemComparator.kt (hunk 5 above)
Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/ui/RecyclerItemComparator.kt]: done

File: app/src/main/java/com/example/trempel/ui/RecyclerItem.kt (hunk 4 above)
Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/ui/RecyclerItem.kt]: done

File: app/src/main/java/com/example/trempel/ui/DiffCallBack.kt (hunk 3 above)
Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/ui/DiffCallBack.kt]: done

File: app/src/main/java/com/example/trempel/ui/DataBindingRecyclerAdapter.kt (hunk 2 above)
Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/ui/DataBindingRecyclerAdapter.kt]: done

**Review by DaryaErofeeva — APPROVED** [review]

**Pull request merged**